server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java

```diff
@@ -419,21 +419,13 @@ private String decodedRepositoryUrl(Request request) throws UnsupportedEncodingE
         return URLDecoder.decode(request.params("encodedUrl"), StandardCharsets.UTF_8.displayName());
     }
 
-    private List<AdditionalFields> extractAdditionalFields(String additionalFieldsParam) throws IllegalArgumentException {
-        List<AdditionalFields> additionalFields = new ArrayList<>();
-        Map<String,AdditionalFields> parametersEquivalent = new HashMap<>();
-        parametersEquivalent.put("attributes", AdditionalFields.ATTRIBUTES);
-        parametersEquivalent.put("perRecipientsHeaders", AdditionalFields.PER_RECIPIENTS_HEADERS);
-        parametersEquivalent.put("headers", AdditionalFields.HEADERS);
-        parametersEquivalent.put("body", AdditionalFields.BODY);
-        parametersEquivalent.put("messageSize", AdditionalFields.MESSAGE_SIZE);
-
-        for (String requestedField : additionalFieldsParam.split(",")) {
-            if (requestedField.isEmpty()) {
-                continue;
-            }
-            if (parametersEquivalent.containsKey(requestedField)) {
-                additionalFields.add(parametersEquivalent.get(requestedField));
+    private List<AdditionalField> extractAdditionalFields(String additionalFieldsParam) throws IllegalArgumentException {
+        List<AdditionalField> additionalFields = new ArrayList<>();
+
+        for (String requestedField : Splitter.on(',').trimResults().omitEmptyStrings().split(additionalFieldsParam)) {
+            Optional<AdditionalField> correspondingField = AdditionalField.find(requestedField);
+            if (correspondingField.isPresent()) {
+                additionalFields.add(correspondingField.get());
```

Comment:
That is a side effect.

Side effects are easy to write but hard to understand and not viable in the long run.
 
Please do take time to rewritte this with Java 8 streams.